packages/core/saved-objects/core-saved-objects-migration-server-internal/src/model/model.ts

```diff
@@ -1448,8 +1448,8 @@ export const model = (currentState: State, resW: ResponseType<AllActionStates>):
         };
       } else if (isTypeof(left, 'compared_mappings_changed')) {
         const rootFields = Object.keys(getBaseMappings().properties);
-        const updatedRootFields = left.updatedHashes.filter((field) => rootFields.includes(field));
-        const updatedTypesQuery = Option.fromNullable(buildPickupMappingsQuery(left.updatedHashes));
+        const updatedRootFields = left.updatedTypes.filter((field) => rootFields.includes(field));
+        const updatedTypesQuery = Option.fromNullable(buildPickupMappingsQuery(left.updatedTypes));
```

Comment:
The `modelVersion` logic is currently not including the root fields, so with the PR we're not storing model versions for them in the `_meta`.

I have an update on the pipes to preserve existing behavior, and what I propose is to deep compare the mappings objects for the root fields (which are pretty simple). This way, the updated types will handle comparison of SO types through their `modelVersions` and comparison of root fields through deep compare of their mappings.
It will emit 2 different _Left_ responses depending on which of them has changed (`root_fields_changed` vs `types_changed`), which in turn will simplify the logic in `model.ts`.

WDYT?

Reply:
Seems good to me